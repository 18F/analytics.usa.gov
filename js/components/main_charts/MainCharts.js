import React from "react";
import PropTypes from "prop-types";

import RealtimeVisitors from "./RealtimeVisitors";
import LocationsAndLanguages from "./LocationsAndLanguages";
import Sessions30Days from "./Sessions30Days";
import Visitors30Days from "./Visitors30Days";
import Engagement from "./Engagement";
import TrafficSources from "./TrafficSources";
import DeviceDemographics from "./DeviceDemographics";
import TopPages from "./TopPages";
import TopDownloads from "./TopDownloads";

/**
 * Contains charts and other data visualizations for the main page of the site.
 * This component is mainly laying out the structure for the page and passes
 * props necessary for getting data and displaying visualizations to child
 * components.
 *
 * This component is using USWDS grid classes and expects it's parent element to
 * have class 'grid-row'
 *
 * @param {String} dataURL the URL of the base location of the data to be
 * displayed In production this is proxied and redirected to the S3 bucket URL
 * by NGINX.
 * @param {String} dataPrefix the path to add to the base URL to find data for
 * the current agency.
 * @param {String} agency the display name for the current agency.
 */
function MainCharts({
  dataURL,
  dataPrefix,
  agency = "U.S. Federal Government",
}) {
  const dataHrefBase = `${dataURL}/${dataPrefix}`;

  return (
    <>
      <section id="main_data" className="desktop:grid-col-8">
        <article className="chart-realtime">
          <RealtimeVisitors dataHrefBase={dataHrefBase} agency={agency} />
        </article>

        <article className="section">
          <LocationsAndLanguages dataHrefBase={dataHrefBase} />
        </article>

        <article className="section">
          <div className="section__headline">
            <h2>Historical Data and Trends</h2>
          </div>
        </article>

        <article className="section">
          <div className="section__headline">
            <h3>Daily Sessions in the Past 30 Days</h3>
          </div>
          <Sessions30Days dataHrefBase={dataHrefBase} />
        </article>

        <article className="section">
          <section className="section__subheader">
            <Visitors30Days dataHrefBase={dataHrefBase} />
          </section>
        </article>

        <article className="section">
          <section className="section__rate">
            <div className="grid-row">
              <Engagement dataHrefBase={dataHrefBase} />
            </div>
          </section>
        </article>

        <article className="section">
          <div className="section__headline">
            <h2>Top Traffic Sources in the Last 30 Days</h2>
          </div>

          <section className="section__chart">
            <div className="grid-row">
              <TrafficSources dataHrefBase={dataHrefBase} />
            </div>
          </section>
        </article>

        <article className="section">
          <div className="section__headline">
            <h2>User Device Demographics in the Last 30 Days</h2>
          </div>

          <section className="section__chart">
            <div className="grid-row">
              <DeviceDemographics dataHrefBase={dataHrefBase} />
            </div>
          </section>
        </article>
      </section>

      <section id="secondary_data" className="desktop:grid-col-4">
        <section>
          <TopPages dataHrefBase={dataHrefBase} />
        </section>
        <section className="top-downloads">
          <div className="top-downloads__headline">
            <h3>Top Downloads Yesterday</h3>
          </div>

          <h4>
            <em>Total file downloads yesterday on {agency} hostnames.</em>
          </h4>
          <TopDownloads dataHrefBase={dataHrefBase} />
        </section>
      </section>
    </>
  );
}

MainCharts.propTypes = {
  dataURL: PropTypes.string.isRequired,
  dataPrefix: PropTypes.string.isRequired,
  agency: PropTypes.string,
};

export default MainCharts;
