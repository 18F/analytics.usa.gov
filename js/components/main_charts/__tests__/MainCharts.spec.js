import renderer from "react-test-renderer";
import MainCharts from "../MainCharts";

describe("MainCharts", () => {
  it("renders", () => {
    const component = renderer.create(
      <MainCharts
        dataURL="http://www.example.com"
        dataPrefix="interior"
        agency="Department of Interior"
      />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
