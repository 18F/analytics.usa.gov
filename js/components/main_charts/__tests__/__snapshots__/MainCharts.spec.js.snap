// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainCharts renders 1`] = `
[
  <section
    className="desktop:grid-col-8"
    id="main_data"
  >
    <article
      className="chart-realtime"
    >
      <section
        className="chart-realtime__visitors-count"
        data-refresh="15"
      >
        <div
          className="grid-row"
        >
          <h2
            className="chart-realtime__current-visitors data grid-col-12"
          >
            ...
          </h2>
        </div>
        <div
          className="grid-row"
        >
          <div
            className="chart-realtime__description desktop:grid-col-8 desktop:grid-offset-2"
          >
            people on 
            Department of Interior
             websites and apps in the last 30 minutes
          </div>
        </div>
      </section>
    </article>
    <article
      className="section locations-and-languages-section"
    >
      <div
        className="section__headline"
      >
        <h2>
          User Locations and Languages in the Last 30 Minutes
        </h2>
      </div>
      <div
        className="section__chart grid-row"
      >
        <section
          className="desktop:grid-col-4 padding-2 bar-chart-component"
          id="cities"
        >
          <h4>
            Cities
          </h4>
          <figure
            data-refresh="15"
            data-source="http://www.example.com/interior/top-cities-realtime.json"
            id="chart_top-cities-90-days"
          >
            <div
              className="data bar-chart"
            />
          </figure>
        </section>
        <section
          className="desktop:grid-col-4 padding-2 bar-chart-component"
          id="countries"
        >
          <h4>
            Countries
          </h4>
          <div>
            <figure
              id="chart_us"
            >
              <div
                className="data bar-chart"
              />
            </figure>
            <figure
              className="hide"
              id="chart_us_and_territories"
            >
              <h4>
                United States & Territories
              </h4>
              <div
                className="data bar-chart"
              />
            </figure>
            <figure
              className="hide"
              id="chart_countries"
            >
              <h4>
                International
              </h4>
              <div
                className="data bar-chart"
              />
            </figure>
          </div>
        </section>
        <section
          className="desktop:grid-col-4 padding-2 bar-chart-component"
          id="languages"
        >
          <h4>
            Languages
          </h4>
          <figure
            data-block="languages"
            id="chart_top-languages"
          >
            <div
              className="data bar-chart"
            />
          </figure>
        </section>
      </div>
    </article>
    <article
      className="section"
    >
      <div
        className="section__headline"
      >
        <h2>
          Historical Data and Trends
        </h2>
      </div>
    </article>
    <article
      className="section sessions-30-days-section"
    >
      <div
        className="section__headline"
      >
        <h3>
          Daily Sessions in the Past 30 Days
        </h3>
      </div>
      <div
        className="section__chart"
        data-refresh="15"
        id="time_series"
      >
        <svg
          className="data time-series"
        />
      </div>
    </article>
    <article
      className="section"
    >
      <section
        className="section__subheader"
      >
        <div>
          There were
           
          <span
            className="data"
            id="total_visitors"
          >
            ...
          </span>
           
          sessions in the last 30 days.
        </div>
      </section>
    </article>
    <article
      className="section"
    >
      <section
        className="section__rate"
      >
        <div
          className="grid-row"
        >
          <section
            className="desktop:grid-col-6"
            id="average_engagement_time"
          >
            <div
              className="grid-row"
            >
              <h4
                className="grid-col-12"
              >
                <a
                  href="/definitions#report_historical_average_engagement_time"
                >
                  Average Engagement Time Per Session
                </a>
              </h4>
            </div>
            <div
              className="grid-row"
            >
              <span
                className="grid-col-12"
              >
                <div>
                  <div
                    className="data"
                  />
                </div>
              </span>
            </div>
          </section>
          <section
            className="desktop:grid-col-6"
            id="engagement_rate"
          >
            <div
              className="grid-row"
            >
              <h4
                className="grid-col-12"
              >
                <a
                  href="/definitions#report_historical_engagement_rate"
                >
                  Percent of Engaged Sessions
                </a>
              </h4>
            </div>
            <div
              className="grid-row"
            >
              <span
                className="data grid-col-12"
              >
                <div>
                  <div
                    className="data"
                  />
                </div>
              </span>
            </div>
          </section>
        </div>
      </section>
    </article>
    <article
      className="section traffic-sources-section"
    >
      <div
        className="section__headline"
      >
        <h2>
          Top Traffic Sources in the Last 30 Days
        </h2>
      </div>
      <section
        className="section__chart"
      >
        <div
          className="grid-row"
        >
          <section
            className="desktop:grid-col-6 bar-chart-component"
            id="session_channel_groups"
          >
            <h4>
              Top Channels
            </h4>
            <figure
              id="chart_session_channel_group"
            >
              <div
                className="data bar-chart"
              />
            </figure>
          </section>
          <section
            className="desktop:grid-col-6 bar-chart-component"
            id="session_source_mediums"
          >
            <h4>
              Top Sources/Media
            </h4>
            <figure
              id="chart_session_source_medium"
            >
              <div
                className="data bar-chart"
              />
            </figure>
          </section>
        </div>
      </section>
    </article>
    <article
      className="section device-demographics-section"
    >
      <div
        className="section__headline"
      >
        <h2>
          User Device Demographics in the Last 30 Days
        </h2>
      </div>
      <section
        className="section__chart"
      >
        <div
          className="grid-row"
        >
          <section
            className="desktop:grid-col-4 bar-chart-component"
            id="devices"
          >
            <h4>
              Devices
            </h4>
            <figure
              id="chart_device_types"
            >
              <div
                className="data bar-chart"
              />
            </figure>
          </section>
          <section
            className="desktop:grid-col-4 bar-chart-component"
            id="browsers"
          >
            <h4>
              Web Browsers
            </h4>
            <figure
              id="chart_browsers"
            >
              <div
                className="data bar-chart"
              />
            </figure>
          </section>
          <section
            className="desktop:grid-col-4 bar-chart-component"
            id="operating_systems"
          >
            <h4>
              Operating Systems
            </h4>
            <div>
              <figure
                id="chart_os"
              >
                <div
                  className="data bar-chart"
                />
              </figure>
              <figure
                className="hide"
                data-scale-to-parent="true"
                id="chart_windows"
              >
                <h4>
                  Windows
                </h4>
                <div
                  className="data bar-chart"
                />
              </figure>
            </div>
          </section>
        </div>
      </section>
    </article>
  </section>,
  <section
    className="desktop:grid-col-4"
    id="secondary_data"
  >
    <section
      className="top-pages-and-downloads"
    >
      <div
        className="top-pages-and-downloads__headline"
      >
        <h2>
          Top 
          30
           Web Pages and App Screens
        </h2>
      </div>
      <div
        className="react-tabs"
        data-rttabs={true}
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <ul
          className="usa-button-group usa-button-group--segmented"
          role="tablist"
        >
          <li
            aria-controls="panel:r0:0"
            aria-disabled="false"
            aria-selected="true"
            className="usa-button-group__item react-tabs__tab--selected"
            data-rttab={true}
            id="tab:r0:0"
            role="tab"
            tabIndex="0"
          >
            <button
              className="usa-button"
              type="button"
            >
              30 mins
            </button>
          </li>
          <li
            aria-controls="panel:r0:1"
            aria-disabled="false"
            aria-selected="false"
            className="usa-button-group__item"
            data-rttab={true}
            id="tab:r0:1"
            role="tab"
            tabIndex={null}
          >
            <button
              className="usa-button"
              type="button"
            >
              7 days
            </button>
          </li>
          <li
            aria-controls="panel:r0:2"
            aria-disabled="false"
            aria-selected="false"
            className="usa-button-group__item"
            data-rttab={true}
            id="tab:r0:2"
            role="tab"
            tabIndex={null}
          >
            <button
              className="usa-button"
              type="button"
            >
              30 days
            </button>
          </li>
        </ul>
        <div
          aria-labelledby="tab:r0:0"
          className="react-tabs__tab-panel react-tabs__tab-panel--selected"
          id="panel:r0:0"
          role="tabpanel"
        >
          <section
            className="top-pages"
          >
            <p>
              <strong>
                People
              </strong>
               on a
               
              <strong>
                single, specific page or app screen
              </strong>
               in the last 30 minutes. Hostnames are not currently reported in real-time, so only page title and screen name information is available.
            </p>
            <figure
              className="top-pages__realtime-bar-chart"
              data-refresh="15"
              data-source="http://www.example.com/interior/top-pages-realtime.json"
            >
              <div
                className="data bar-chart"
              />
            </figure>
          </section>
          <section
            className="top-downloads"
          >
            <div
              className="top-downloads__headline"
            >
              <h3>
                Top 
                20
                 Downloads Yesterday
              </h3>
            </div>
            <h4>
              <em>
                Total file downloads yesterday on 
                Department of Interior
                 hostnames.
              </em>
            </h4>
            <figure
              className="top-downloads__bar-chart"
            >
              <div
                className="data bar-chart"
              />
            </figure>
          </section>
        </div>
        <div
          aria-labelledby="tab:r0:1"
          className="react-tabs__tab-panel"
          id="panel:r0:1"
          role="tabpanel"
        />
        <div
          aria-labelledby="tab:r0:2"
          className="react-tabs__tab-panel"
          id="panel:r0:2"
          role="tabpanel"
        />
      </div>
    </section>
  </section>,
]
`;
